common/djangoapps/student/roles.py

```diff
@@ -49,7 +49,7 @@ def has_role(self, role, course_id, org):
         """
         return any(
             access_role.role == role and
-            access_role.course_id == course_id and
+            unicode(access_role.course_id) == unicode(course_id) and
```

Comment:
second option is block this PR and create small PR against 
https://github.com/edx/ccx-keys/blob/master/ccx_keys/locator.py

and override __eq__ method. Because we dont have this method thats why when i compare 2 ccx locator objects it always returns false

After that resume this PR
http://stackoverflow.com/questions/1227121/compare-object-instances-for-equality-by-their-attributes-in-python